src/coord/rmr/conn.c

```diff
@@ -432,20 +432,17 @@ static int MRConn_SendAuth(MRConn *conn) {
     // Take the GIL before calling the internal function getter
     RedisModule_ThreadSafeContextLock(RSDummyContext);
     const char *internal_secret = RedisModule_GetInternalSecret(RSDummyContext, &len);
-    RedisModule_ThreadSafeContextUnlock(RSDummyContext);
-
-    sds secret = sdsnewlen(internal_secret, len);
-
-    if (redisAsyncCommand(conn->conn, MRConn_AuthCallback, conn,
-        "AUTH %s %s", INTERNALAUTH_USERNAME, secret) == REDIS_ERR) {
-      sdsfree(secret);
+    // Create a local copy of the secret so we can release the GIL.
+    int status = redisAsyncCommand(conn->conn, MRConn_AuthCallback, conn,
+        "AUTH %s %b", INTERNALAUTH_USERNAME, internal_secret, len);
+    if (status == REDIS_ERR) {
       MRConn_SwitchState(conn, MRConn_ReAuth);
-      return REDIS_ERR;
     }
-    sdsfree(secret);
-    return REDIS_OK;
+    RedisModule_ThreadSafeContextUnlock(RSDummyContext);
+    return status;
```

Comment:
We sure that the status return from `redisAsyncCommand` are the same as `REDIS_OK`/`REDIS_ERR`?

Reply:
Our whole connection mechanism uses them..
I agree it's not the best practice but let's change it later in a small refactor